### PR TITLE
08.02.01 Add support for deleting images from books

### PR DESCRIPTION
**Pull request by imswebra**
To be merged after PR #56.

Very small change, just adds functionality to the image delete button. Somewhat rudimentary implementation until the proper UI for the add / edit activities is implemented.

### CODE REVIEW

**Comment by mitchadam** [comment]
> Very small change, just adds functionality to the image delete button. Somewhat rudimentary implementation until the proper UI for the add / edit activities is implemented.

What functionality isn't in the current UI?

File: src/app/src/main/java/com/example/bookmark/AddBookActivity.java [hunk 1]
```diff
@@ -18,10 +23,13 @@
  * Need to hook up to DB
  * @author Mitch Adam.
  */
-public class AddBookActivity extends BackButtonActivity {
+public class AddBookActivity extends AppCompatActivity implements ImageSelectDialogFragment.ImageSelectListener {
```
Comment (ryankortbeek) [inline src/app/src/main/java/com/example/bookmark/AddBookActivity.java:26]: Does this screen have a back button?

Comment (ryankortbeek) [reply src/app/src/main/java/com/example/bookmark/AddBookActivity.java:26]: Never-mind I see the startActivityForResult below - ignore these comments

**Review by ryankortbeek — APPROVED** [review]
Everything looks good!

**Comment by imswebra** [comment]
> What functionality isn't in the current UI?
@mitchadam 

In general? Imo the add book and edit book layouts should be consolidated into a single layout (and the activity classes should similarly extend from a common parent, but that's a different conversation). This new layout would be a rebuild of either using material.io components and reference common colors from `colors.xml`. We also probably want to move to a proper background for the addPhotoButton instead of using a drawable in the foreground for when there is no image selected. I'd also like to see the addPhotoButton be dynamic in size to fill any remaining space on screen (having the done / delete buttons always in the same place relative to the bottom edge) and to match the aspect ratio of its contents.

With respect to my "rudimentary implementation" comment though, I was moreso just referencing the lines where I call `addPhotoButton.setImageDrawable(ContextCompat.getDrawable(this, android.R.drawable.ic_menu_add));`. Once we get around to UI fine-tuning and we have a proper background for addPhotoButton with the plus button embedded in it, we should be clearing the addPhotoButton image instead of replacing it on delete.

**Pull request merged**